src/ast/token.rs

```diff
@@ -0,0 +1,171 @@
+use std::fmt::Display;
+use std::fmt;
+
+#[derive(Debug, PartialEq, Eq)]
+pub struct TokenizerError {
+    pub error_type: ErrorType,
+}
+
+#[derive(Debug, PartialEq, Eq)]
+pub enum ErrorType {
+    NumberError
+}
+
+impl Display for ErrorType {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            ErrorType::NumberError => write!(
+                f,
+                "tokenized number error",
+            )
+        }
+    }
+}
+
+impl Display for TokenizerError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(
+            f,
+            "{}",
+            self.error_type
+        )
+    }
+}
+
+impl std::error::Error for TokenizerError {}
+
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub enum Token {
+    /// An end-of-file marker, not a real token
+    EOF,
+    /// A keyword (like SELECT) or an optionally quoted SQL identifier
+    Word(Word),
+    /// An numeric literal
+    Number(String),
+    /// Single quoted string: i.e: 'string'
+    SingleQuotedString(String),
+    /// Whitespace (space, tab, etc)
+    Whitespace(Whitespace),
+    /// Equality operator `=`
+    Eq,
+    /// Not Equals operator `<>` (or `!=` in some dialects)
+    Neq,
+    /// Less Than operator `<`
+    Lt,
+    /// Greater Than operator `>`
+    Gt,
+    /// Less Than Or Equals operator `<=`
+    LtEq,
+    /// Greater Than Or Equals operator `>=`
+    GtEq,
+    /// Period (used for compound identifiers or projections into nested types)
+    Period,
+    /// Comma `,`
+    Comma,
+    /// SemiColon `;` used as separator for COPY and payload
+    SemiColon,
```

Comment:
Should we add quote character "`". This is used in Bigquery.

Reply:
leave it for later